app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java

```diff
@@ -517,4 +510,32 @@ private boolean isPermissionGranted() {
         }
         return true;
     }
+
+    @Override
+    public void onPDFCreationStarted() {
+        mMaterialDialog = createAnimationDialog(mActivity);
+        mMaterialDialog.show();
+    }
+
+    @Override
+    public void onPDFCreated(boolean success, String path) {
+        if (mMaterialDialog != null && mMaterialDialog.isShowing())
+            mMaterialDialog.dismiss();
+        if (!success) {
+            showSnackbar(mActivity, R.string.error_occurred);
+            mMorphButtonUtility.morphToGrey(mCreateTextPdf, mMorphButtonUtility.integer());
+            mCreateTextPdf.setEnabled(false);
+            mTextFileUri = null;
+            mButtonClicked = 0;
+            return;
+        }
+        final String finalMPath = mPath;
```

Comment:
what is the need of `finalMPath`?

Reply:
it is used. Please see.

Reply:
I mean I understand there is no need of that

Reply:
But it was present in that fashion earlier so I didn't change that.

Reply:
then remove it